compiler-rt/lib/mpk_untrusted/alloc_site_handler.h

```diff
@@ -104,6 +113,10 @@ class AllocSiteHandler {
   std::set<AllocSite> fault_set;
   // Thread safety mutex
   std::mutex mx;
+  // Mapping of thread-id to saved pkey
+  std::map<pid_t, PKeyInfo> pid_key_map;
+  // MPK map safety mutex
+  std::mutex px;
```

Comment:
what is the difference between this and mx? I cannot readily see from the comments. also, why are these `safety` mutexs? mutual exclusion is always a safety property. Is this an exclusive lock on pid_key_map? then what does mx lock?

Reply:
Changed naming and comments to better indicate intended usage of both mutexes.